Assignment 3 Instructions.Rmd

```diff
@@ -153,17 +153,150 @@ K7 <- dplyr::count(K4, cluster)
 
 Look at the number of people in each cluster, now repeat this process for 3 rather than 2 clusters. Which cluster grouping do you think is more informative? Write your answer below:
 
+```{r}
+fit_3cluster<-kmeans(K3, 3)
+
+fit_3cluster$cluster
+
+C_31<-data.frame(K3, fit_3cluster$cluster)
+
+names(C_31)<-c("1","2","3","4","5","cluster")
+
+C_32<-tidyr::gather (C_31, "week", "motivation", 1:5)
+
+C_33<-C_32 %>% group_by(week, cluster)
+
+C_33<-summarise(C_33, avg=mean(motivation))
+
+C_33$week<-as.numeric(C_33$week)
+
+C_33$cluster<-as.factor(C_33$cluster)
+
+ggplot(C_33, aes(week, avg, colour = cluster))+geom_point() + geom_line() + xlab("Week") + ylab("Average Motivation")
+
+C_34<-dplyr::count(C_31, cluster)
+
+# Personally, I think the three clusters looks better than 2 clusters. Because the plot with three clusters shows 3 different patterns for students. None of these patterns show similar trend to each other. Therefore, I think this plot contains more information than the 2-cluster plot.
```

Comment:
It isn't really about whether a plot shows more information, but rather whether the clusters are meaningful. Can you think of a reason that three clusters makes more sense than two clusters? Some people think the simplest explanation is always the best (Occam's Razor).

Reply:
I usually decide the number of clusters by looking at the total variance explained and interpretability. I think the visualization is a totally new method to me so I think we need to interpret the plot along with the questions and original data. I think the more clusters we have, the higher amount of variance would be explained. The important part is that whether these clusters are meaningful. According to the patterns, I think the three clusters show some significant distinctions from each other. Therefore, I chose to keep the three clusters.